More new types for 1.0.x minor release

bwiernik opened the pull request:
Separate PRs coming for variables and terms. 

Compiled from discussions on csl-evolution, zotero-bits, schema, and documentation. Also from the CSL-M spec.

1. `performance` is needed for live performances. Arts and musicology styles have distinct formatting rules for live performances versus conference-style presentations
2. `periodical` is for references to whole periodicals (e.g., a run of a journal, a special issue)
3. `collection` is for museum or archival collections. I believe that this would also cover citation of exhibitions/exhibit catalogues (@adam3smith?)
4. `hearing` and `regulation` are legal types with distinct citation needs that `bill` or `legislation` (e.g., Congressional hearings).
5. `classic` refers to ancient works with unique citation forms versus modern works (widely used in classical and biblical studies, for example). 

Two CSL-M types I omitted:
1. `video`:
    - Defined as "The video type is appropriate for video works that are not disseminated through an access-restricted distribution channel. For example, content hosted on YouTube should be set to `video`, while a DVD release of “The Wizard of Oz” should be set to `motion_picture`."
    - I don't think this is a meaningful distinction, and I'm not sure that such a distinction is required by citation styles.
1. `gazette`:
    - A specific pubication format for some legal items
    - Appears to be superseded by a `gazette-flag` variable that might be assigned for types `legislation` or `regulation`.

Closes https://github.com/citation-style-language/csl-evolution/issues/22
Closes https://github.com/citation-style-language/schema/issues/22

Addresses https://github.com/citation-style-language/schema/issues/95

I've tagged zotero-bits issues with a `zotero` tag to indicate that all that remains on them is Zotero implementation.

bdarcus commented:
I requested reviewed from a few people. Hoping they can do this sometime during the coming week.

bwiernik commented:
Where do we stand on closing the zotero-bits issues that also entail Zotero changes?

bdarcus commented:
Obviously we wouldn't want to close them, but we could link from here.

We should also probably transfer those to Zotero?

denismaier commented:
> Obviously we wouldn't want to close them, but we could link from here. We should also probably transfer those to Zotero?

We've had some discussion about this a couple of weeks ago, right?

Perhaps we can just transfer the whole repo to Zotero once we've dealt with the CSL part...

bwiernik commented:
@bdarcus Could you give me editing rights To that repo so I can add a “CSLdone” tag to issues as we resolve them?

bdarcus commented:
I just gave you "triage" rights, which I think works here. Let me know if not.

bwiernik commented:
Updated the commit comment to close issues. For zotero-bits issues, I added `zotero` tags.

(By the way, for the extended commit comments, you can edit them directly on GitHub, rather than having to use git commands.)

bdarcus commented:
I don't see any changes ATM. Latest commit says yesterday.

bwiernik commented:
I just updated the commit comment, didn't make a new commit. The bottom of the commit comment now closes 2 issues.

rmzelle commented on csl-types.rnc:
```diff
@@ -10,12 +10,15 @@ div {
     | "broadcast"
     | "chapter"
+      "classic"
+      "collection"
```
@bwiernik, these new entries all need to be preceded by a pipe ("|"), which IIRC means OR in RNC.

(we should have some CI to make sure the schema is valid)

bdarcus replied:
Tip - easiest way to validate RNC is to use Trang to convert it to RNG. If it converts without error, it's valid.

rmzelle replied:
> (we should have some CI to make sure the schema is valid)

Do we have a ticket for this?

bdarcus replied:
Just added #197

bwiernik commented:
Ah, sorry. The vertical bars overlapped with my editor’s indentation guides and I didn’t notice.

rmzelle commented:
> Perhaps we can just transfer the whole repo to Zotero once we've dealt with the CSL part...

Dan Stillman actually asked for this here: https://github.com/citation-style-language/zotero-bits/issues/73#issuecomment-591806524

bwiernik commented:
Yeah, so once we've finished tagging everything with `zotero`, we can transfer it to them.

bdarcus commented on csl-data.json:
```diff
@@ -17,12 +17,14 @@
                     "book",
                     "broadcast",
                     "chapter",
```
adding "classic" here?

bdarcus commented on csl-data.json:
```diff
@@ -36,11 +38,13 @@
+                    "regulation",
                     "report",
                     "review",
                     "review-book",
```
"software"?

bdarcus commented on csl-data.json:
```diff
@@ -36,11 +38,13 @@
                     "personal_communication",
```
"periodical"?

bdarcus commented:
OK, merged. Thanks.

Note, I simplified the commit message, since that info is included here.
